src/main/java/com/zextras/SlcwClient.java

```diff
@@ -1,9 +1,147 @@
 package com.zextras;
 
-/** FIXME add actual documentation when it will be implemented */
+import com.unboundid.ldap.sdk.*;
+import com.zextras.operations.ldap.LdapOperationExecutor;
+import com.zextras.operations.OperationResult;
+import com.zextras.persistence.converters.SlcwConverter;
+import com.zextras.persistence.mapping.SlcwEntry;
+import com.zextras.persistence.mapping.SlcwMapper;
+import com.zextras.utils.PropertyBuilder;
+import com.zextras.utils.ObjectFactory;
+
+/**
+ * Main entrypoint for the library.
+ */
 public class SlcwClient {
-  // FIXME remove this when actually implementing the codebase :)
-  public static void main(String[] args) {
-    System.out.println("Hello World!");
+
+  //todo connection factory
+  private LDAPConnection connection;
+  private final String baseDn;
+  private final SlcwMapper mapper = new SlcwMapper();
+  private final PropertyBuilder builder = new PropertyBuilder();
+
+  /**
+   * Creates a client with an opened connection.
+   *
+   * @param connection an opened connection to the server.
+   * @param baseDn     the starting point on the server.
+   */
+  public SlcwClient(LDAPConnection connection, String baseDn) {
+    this.baseDn = baseDn;
+    this.connection = connection;
+  }
+
+  /**
+   * Creates a client, authenticate a user and change the identity of the client connection.
+   *
+   * @param host     a network layer host address.
+   * @param port     a port on a host that connects it to the storage system.
+   * @param bindDN   a Username used to connect to the server.
+   * @param password a secret word or phrase that allows access to the server.
+   * @param baseDn   the starting point on the server.
+   */
+  public SlcwClient(String host, int port, String bindDN, String password, String baseDn) {
+    this.baseDn = baseDn;
+    initialize(host, port, bindDN, password);
+  }
+
+  private void initialize(String host, int port, String bindDN, String password) {
+    try {
+      connection = new LDAPConnection(host, port, bindDN, password);
+    } catch (LDAPException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Gets an object from a record stored in the structure, otherwise throws an exception.
+   *
+   * @param id    a unique identifier that marks that particular record as unique from every other
+   *              record.
+   * @param clazz type of the class which represents an object that you wanted to get. (ex.
+   *              User.class)
+   * @param <T>   is a conventional letter that stands for "Type".
+   * @return an object of the given class.
+   */
+  public <T> T getById(String id, Class<T> clazz) {
+    T object = ObjectFactory.newObject(clazz);
+    SlcwEntry entry = new SlcwEntry(baseDn);
+
+    mapper.map(object, entry);
+
+    entry.getId().setFiledValue(id);
+    entry.setDn(builder.buildDn(entry, object));
+    entry.setFilter(builder.buildFilter(entry));
```

Comment:
I'm checking usage of SlcwClient api as well the SlcwEntry.
About setting dn and filter, do you think we can perform it in the mapper, since also update and add method do the same thing?

Reply:
dn and filter depends on method
add and update same but differs from delete and get
if we want to set it inside mapper we have to provide information about method which calls it
what do you think about it?

Reply:
Do they? Asking because getById delete, add, update all do the same operations:
- map entry
- setFilter
- setDn
setFilter and setDn do not have any specifiic logic that depends on called method

Reply:
sorry i forgot i moved that logic in operation executor
sorry

Reply:
Ok, then you can just remove entry.setDn and entry.SetFilter right? 
Also I would rethink a bit dn and filter fields, IMHO:
- baseDn can be a SlcwEntry field, but would make it final. This way you do not risk moving things around by mistake. E.g.: what happens if I get an Entry, then change it's baseDn and then call add?
- filter should be something only SlcwClient understands. So the client is able to generate a filter/request based on given entry fields

Reply:
i moved entry.setDn and entry.SetFilter to mapper
also baseDN final now

Reply:
To follow up what we talked about yesterday: avoid side effects like passing an object to a method and changing its state